frontend/src/pages/notebookController/screens/server/usePreferredNotebookSize.ts

```diff
@@ -1,29 +1,33 @@
 import * as React from 'react';
-import AppContext from '../../../../app/AppContext';
+import { useAppContext } from '../../../../app/AppContext';
 import { useNotebookUserState } from '../../../../utilities/notebookControllerUtils';
+import { DEFAULT_NOTEBOOK_SIZES } from '../../const';
+import { NotebookSize } from '../../../../types';
 
-export const usePreferredNotebookSize = (): [
-  selectedSize: string,
-  setSelectedSize: (size: string) => void,
-] => {
-  const { dashboardConfig } = React.useContext(AppContext);
+export const usePreferredNotebookSize = (): {
+  selectedSize: string;
+  setSelectedSize: (size: string) => void;
+  sizes: NotebookSize[];
+} => {
+  const { dashboardConfig } = useAppContext();
+  const sizes = dashboardConfig?.spec?.notebookSizes || DEFAULT_NOTEBOOK_SIZES;
   const currentUserState = useNotebookUserState();
 
   let defaultSize = '';
-  if (dashboardConfig?.spec.notebookSizes) {
-    if (currentUserState.lastSelectedSize) {
-      const size = dashboardConfig.spec.notebookSizes.find(
-        (notebookSize) => notebookSize.name === currentUserState.lastSelectedSize,
-      );
-      if (size) {
-        defaultSize = size.name;
-      } else {
-        defaultSize = dashboardConfig.spec.notebookSizes[0].name;
-      }
+  if (currentUserState.lastSelectedSize) {
+    const size = sizes.find(
+      (notebookSize) => notebookSize.name === currentUserState.lastSelectedSize,
+    );
+    if (size) {
+      defaultSize = size.name;
     } else {
-      defaultSize = dashboardConfig.spec.notebookSizes[0].name;
+      defaultSize = sizes[0].name;
```

Comment:
Can we also log something into the console here? I feel like we should just note that something funky happened trying to get it. That way if a report comes in we can ask for the console and just say "ah yeah, your default somehow was removed / renamed" rather than just try to reproduce it.